tests/samples/schema.org/product-ref.json

```diff
@@ -0,0 +1,69 @@
+[
+    {
+        "type": "http://schema.org/Product",
+        "properties": {
+            "referenced_product": {
+                "type": "http://schema.org/Product",
+                "properties": {
+                    "name": "REFERENCED PRODUCT",
+                    "image": "img-ref.jpg"
+                }
+            },
+            "prop3": "REFERENCED TO INCLUDE PROPERTIES AND ALSO INDIVIDUAL PRODUCT",
+            "image": [
+                "img-2.jpg",
+                "anvil_executive.jpg"
```

Comment:
The order of the images seems reversed to the depth-first traversal order, I wonder if we should care about this.

Reply:
I'm also not sure. Looking at that: https://www.w3.org/TR/microdata/#associating-names-with-items I would say that referenced items should be processed after the current item. Maybe is just as easy as changing the order on which 
`itemref` is processed. I did a quick test and only one test fails... I'll check deeper later.

Reply:
It makes sense to me to change the order in which `itemref` is processed. I have committed a fix.